lib/blockchain/chain.js

```diff
@@ -2034,10 +2034,12 @@ class Chain extends AsyncEmitter {
    */
 
   getProgress() {
-    const start = this.network.genesis.time;
-    const current = this.tip.time - start;
-    const end = util.now() - start - 40 * 60;
-    return Math.min(1, current / end);
+    const time  = util.now();
+    const rate = 271416 / (24 * 60 * 60);
+    const lastTime = 1672074795;
+    const current = this.db.state.tx;
+    const end = 791077253 + (time - lastTime) * rate;
+    return current / end;
```

Comment:
Ok great, this looks right. I think we need to move these constants to `networks.js`, similar to chainparams.cpp in bitcoind:

https://github.com/bitcoin/bitcoin/blob/master/src/chainparams.cpp

Note how each network (main, regtest, testnet, signet) has its own set of constants (grep for `chainTxData`). Also it's important that for regtest its all `0`!